features/370-auth-ui/sdk.md

```diff
@@ -0,0 +1,166 @@
+# SDK
+
+The following new methods are added to access Auth UI.
+
+## React Native
+
+```typescript
+interface LoginOptions {
+  onUserDuplicate?: "abort" | "merge" | "create";
+}
+
+// Open a WebView to login. If it is canceled, the promise will be rejected.
+function loginWithWebUI(redirectURI: string, options?: LoginOptions): Promise<User>;
+
+// Open a WebView to access the given Auth UI path.
+// It interacts with the native cookie storage to ensure
+// the session is attached to the WebView correctly.
+function openWebUI(path: string): Promise<void>;
+```
+
+## Web
```

Comment:
Actually if it is a simple URL with parameters, do we really need to provide Javascript API? Is it really better for developers?

Reply:
The actual URL looks something like `http://192.168.1.127:8081/_auth/_/authorize?client_id=api_key&code_challenge=code_challenge&code_challenge_method=S256&redirect_uri=REDIRECT_URI&response_type=code&scope=openid`. It is not simple enough to let the developer to craft themselves.

Reply:
Noted XD

Reply:
I think documentation-wise, we can still explain both the SDK and the URL.. not "that" complicated and some people may prefer one less JS dependency?

Reply:
I guess the developer cannot get rid of the SDK. The SDK [generates the code challenge](https://tools.ietf.org/html/rfc7636#section-4.2) which is not a trivial task.

Of course the documentation should explain the URL parameters, specifically the difference from the OIDC.